Add unicode chars to formula escape

tomemelko opened the pull request:
This PR consists of two commits:

- A performance improvement for formula escaping, doing `switch` vs `[].includes` resulted in a 4.5x speed up on my machine (https://jsperf.app/wupoqu)
- Addition of unicode equivalent characters for `=`, `+`, `-`, and `@` in formula escaping

tomemelko commented:
@wdavidw do you mind taking a look?

@atlanteh as a CC since I'm changing their code

wdavidw commented:
@tomemelko I am ok with the changes, only the tests must be enriched. Could you do this ?

tomemelko commented:
Hey @wdavidw what needs to be improved in the tests? I can try to do whatever you need, I just need to know what you want.

The other escape characters are just the Unicode equivalents. Apple Numbers does Unicode Normalization which will translate the Unicode characters to their ASCII equivalents, leading to the CSV injection.

For example the Unicode equals sign ('＝') will be translated to the ascii equals sign ('=') and then the formula will be interpreted by Apple Numbers.

You can test this (if you have access to a Mac) by writing the string `＝2+2` to a CSV file and then opening the CSV file in Numbers. You will see `4` in the only cell, meaning the formula has been interpreted.

wdavidw commented:
**About the test:**

The [option.escape_formulas.coffee](https://github.com/adaltas/node-csv/blob/master/packages/csv-stringify/test/option.escape_formulas.coffee) must be updated with the new characters.

**About the documentation:**

> The other escape characters are just the Unicode equivalents.

Yes, I understood, what is relevant is to reference where those characters are documented as relevant/special characters. I personally have no idea about those and I don't wish to blindly commit code which I don't know anything about. For example, you mention that "Apple Numbers does Unicode Normalization which will translate the Unicode characters to their ASCII equivalents". I wish to know where this is written.

arnagar commented:
It might be worth allowing changing the characters to custom characters as it might be useful (for example for people using csv for number reports for negative numbers as - will be escaped so the escape_formulas will be useless for them

tomemelko commented:
# @wdavidw 

## Tests

I could've sworn that I had already updated the test! No wonder I was confused, I really thought I had updated them already. Sorry! It should be updated now.

## Documentation

### CSV Injection

You can read about CSV Injection and the special formula characters here:
- https://owasp.org/www-community/attacks/CSV_Injection
- http://georgemauer.net/2017/10/07/csv-injection.html

### Apple Numbers Unicode Normalization
I cannot find documentation that Apple Numbers is doing Unicode Normalization, but I do have empirical evidence from testing on a Mac. Like I mentioned, if you write the string `＝2+2` (which has the unicode '＝' character) to a CSV file and open it in Numbers it will interpret the formula.

# @arnagar
I could do that as a separate PR, but as it stands right now `escape_formula` will already escape negative numbers, this PR doesn't change that behavior.
